crates/ide-completion/src/context/analysis.rs

```diff
@@ -774,9 +848,14 @@ fn classify_name_ref(
                 ast::TypeBound(_) => TypeLocation::TypeBound,
                 // is this case needed?
                 ast::TypeBoundList(_) => TypeLocation::TypeBound,
-                ast::GenericArg(it) => TypeLocation::GenericArgList(find_opt_node_in_file_compensated(sema, original_file, it.syntax().parent().and_then(ast::GenericArgList::cast))),
+                ast::TypeArg(it) => generic_arg_location(ast::GenericArg::TypeArg(it)),
```

Comment:
This arm should be unnecessary, `ast::GenericArg` will include it for us already